### ISSUE TITLE
Coercion to `CompressedList` discards `mcols`

### ISSUE THREAD

**Issue by LTLA** [Bioconductor/IRanges#53]
Pretty much as it says in the title:

```r
library(IRanges)
X <- list(letters)

Y <- as(X, "CharacterList")
mcols(Y)$foo <- "BAR"
mcols(Y)
## DataFrame with 1 row and 1 column
##           foo
##   <character>
## 1         BAR

Z <- as(Y, "CompressedCharacterList")
mcols(Z)
## NULL
```

<details>
<summary>Session information</summary>

```
R version 4.4.1 Patched (2024-06-20 r86796)
Platform: aarch64-apple-darwin22.6.0
Running under: macOS Ventura 13.6.7

Matrix products: default
BLAS:   /Users/luna/Software/R/R-4-4-branch/lib/libRblas.dylib
LAPACK: /Users/luna/Software/R/R-4-4-branch/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods
[8] base

other attached packages:
[1] IRanges_2.39.0      S4Vectors_0.43.0    BiocGenerics_0.51.0

loaded via a namespace (and not attached):
[1] compiler_4.4.1
```
</details>

**Comment by hpages** [comment]
Oh, we also seem to be loosing them when going from CompressedList to SimpleList:
```
library(IRanges)
x <- IRangesList(A=IRanges(11:15, 20), B=IRanges(5, 7:5), compress=TRUE)
mcols(x)$id <- 101:102
mcols(as(x, "SimpleList"))
# NULL
```
Fixing now...

**Comment by hpages** [comment]
Fixed in **S4Vectors** 0.42.1/0.43.1 and **IRanges** 2.38.1/2.39.1. Thanks for the catch!
